tests/integration_tests/performance/test_snapshot_ab.py

```diff
@@ -0,0 +1,180 @@
+        snapshot = vm.snapshot_full()
+        vm.kill()
+
+        return snapshot
+
+    def sample_latency(self, microvm_factory, guest_kernel, rootfs) -> List[float]:
+        """Collects latency samples for the microvm configuration specified by this instance"""
+        snapshot = self.create_snapshot(microvm_factory, guest_kernel, rootfs)
+
+        values = []
+
+        for _ in range(ITERATIONS):
+            microvm = microvm_factory.build(
+                monitor_memory=False,
+            )
+            microvm.spawn()
+            microvm.restore_from_snapshot(snapshot, resume=True)
+
+            # Check if guest still runs commands.
+            exit_code, _, _ = microvm.ssh.run("true")
+            assert exit_code == 0
+
+            value = 0
+            # Parse all metric data points in search of load_snapshot time.
+            microvm.flush_metrics()
+            metrics = microvm.get_all_metrics()
+            for data_point in metrics:
+                cur_value = data_point["latencies_us"]["load_snapshot"]
+                if cur_value > 0:
+                    value = cur_value / USEC_IN_MSEC
+                    break
+            assert value > 0
+            values.append(value)
+            microvm.kill()
+            shutil.rmtree(Path(microvm.chroot()))
+
+        snapshot.delete()
+        return values
+
+
+@pytest.mark.nonci
+@pytest.mark.parametrize(
+    "test_setup",
+    [
+        SnapshotRestoreTest(mem=128, vcpus=1),
+        SnapshotRestoreTest(mem=1024, vcpus=1),
+        SnapshotRestoreTest(mem=2048, vcpus=2),
+        SnapshotRestoreTest(mem=4096, vcpus=3),
+        SnapshotRestoreTest(mem=6144, vcpus=4),
+        SnapshotRestoreTest(mem=8192, vcpus=5),
+        SnapshotRestoreTest(mem=10240, vcpus=6),
+        SnapshotRestoreTest(mem=12288, vcpus=7),
+        SnapshotRestoreTest(all_devices=True),
+    ],
+    ids=lambda x: x.id,
+)
+def test_restore_latency(
+    microvm_factory, rootfs, guest_kernel_linux_4_14, test_setup, metrics
+):
+    """
+    Restores snapshots with vcpu/memory configuration, roughly scaling according to mem = (vcpus - 1) * 2048MB,
+    which resembles firecracker production setups. Also contains a test case for restoring a snapshot will all devices
+    attached to it.
+
+    We only test a single guest kernel, as the guest kernel does not "participate" in snapshot restore.
+    """
+
+    samples = test_setup.sample_latency(
+        microvm_factory,
+        guest_kernel_linux_4_14,
+        rootfs,
+    )
+
+    metrics.set_dimensions(
+        {
+            "instance": global_props.instance,
+            "cpu_model": global_props.cpu_model,
+            "host_kernel": "linux-" + global_props.host_linux_version,
+            "guest_kernel": guest_kernel_linux_4_14.stem[2:],
+            "rootfs": rootfs.name,
+            "performance_test": "test_restore_latency",
+            **test_setup.dimensions,
+        }
+    )
+
+    metrics.set_property("latency", samples)
```

Comment:
Mh, no, although I can readd that if you want. Having them as properties makes out-of-cloudwatch analysis easier because properties will not be "torn apart" into separate EMFlogs  messages, while metrics are